### PR TITLE
refactor(path,streams): align additional error messages

### PR DESCRIPTION
**Pull request by irbull**
Aligns the error messages in the `streams` and `path` folders.

### CODE REVIEW

**Comment by timreichen** [comment]
I think all error messages with `...must not be...` should reflect the current state, aka `...is...`

File: streams/byte_slice_stream.ts [hunk 1]
```diff
@@ -48,7 +48,9 @@ export class ByteSliceStream extends TransformStream<Uint8Array, Uint8Array> {
     super({
       start: () => {
         if (start < 0) {
-          throw new RangeError("`start` must be greater than 0");
+          throw new RangeError(
+            "Invalid argument: `start` must not be negative",
```
Comment (irbull) [inline streams/byte_slice_stream.ts:52]:
```suggestion
            "Invalid argument: `start` is negative",
```

**Comment by irbull** [comment]
> I think all error messages with `...must not be...` should reflect the current state, aka `...is...`

There are two cases we commonly see:

1. **must not be** where we see things like: `Invalid argument: "start" must not be negative`, so this would change to `Invalid argument: "start" is negative`

But wouldn't this be even better if we had the resolution here? `Invalid argument: `start` must be positive`?

2. **must be** where we see things like: `Path must be absolute, path value "${path}"`. In this case we are saying the current state (path value ${path}) but we are also saying what MUST happen, it must be absolute. The alternative is `Cannot convert to file URL, path value is ${path}` which doesn't really help explain what I need to do.

File: streams/byte_slice_stream.ts (hunk 1 above)
Comment (irbull) [reply streams/byte_slice_stream.ts:52]:
```suggestion
            `Invalid argument, 'start' must >= 0: current value ${start}` ,
```

File: path/_common/from_file_url.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
 export function assertArg(url: URL | string) {
   url = url instanceof URL ? url : new URL(url);
   if (url.protocol !== "file:") {
-    throw new TypeError("Must be a file URL.");
+    throw new TypeError("Must be a file URL");
```
Comment (irbull) [inline path/_common/from_file_url.ts:7]:
```suggestion
    throw new TypeError(`URL must be a file URL: URL protocol is ${url.protocol}`);
```

File: path/from_file_url_test.ts [hunk 3]
```diff
@@ -17,12 +17,12 @@ Deno.test("posix.fromFileUrl()", function () {
   assertThrows(
     () => posix.fromFileUrl("http://localhost/foo"),
     TypeError,
-    "Must be a file URL.",
+    "Must be a file URL",
```
Comment (irbull) [inline path/from_file_url_test.ts:20]:
```suggestion
    "URL must be a file URL: URL protocol is http",
```

**Comment by kt3k** [comment]
> 8. Additional information should describe the current state, if possible, it should also describe the desired state in an affirmative voice:
> 
> ```
> Bad: Cannot compute the square root for x, value must not be negative
> Better: Cannot compute the square root for x: current value is ${x}
> Best: Cannot compute the square root for x, value must be positive: current value is ${x}
> ```

This generally sounds a good idea to me. Can you also send a PR to the style guide?

A minor point: I'm a bit unsure about `,` usage in 'Best' example. Is this comma grammatically fine? What other runtime do for this situation?

**Comment by irbull** [comment]
> This generally sounds a good idea to me. Can you also send a PR to the style guide?
> 
> A minor point: I'm a bit unsure about `,` usage in 'Best' example. Is this comma grammatically fine? What other runtime do for this situation?

Done https://github.com/denoland/std/pull/5757

File: streams/byte_slice_stream.ts (hunk 1 above)
Comment (irbull) [reply streams/byte_slice_stream.ts]: Updated to `'start' must be >= 0: received ${start}`

File: path/_common/from_file_url.ts (hunk 2 above)
Comment (irbull) [reply path/_common/from_file_url.ts]: Updated to `URL must be a file URL: received "${url.protocol}"`

File: streams/buffer.ts [hunk 4]
```diff
@@ -315,7 +315,9 @@ export class Buffer {
       return;
     }
     if (n < 0 || n > this.length) {
-      throw Error("bytes.Buffer: truncation out of range");
+      throw Error(
+        `Error truncating buffer, truncation value "${n}" out of range: value must be between 0 and ${this.length}`,
```
Comment (kt3k) [inline streams/buffer.ts:319]:
```suggestion
        `Cannot truncate the buffer as the truncation value "${n}" is out of range: value must be between 0 and ${this.length}`,
```

Comment (iuioiua) [reply streams/buffer.ts:319]: Seems a little long, no?

Comment (kt3k) [reply streams/buffer.ts:319]: But that is what the style guide now recommend

Comment (timreichen) [reply streams/buffer.ts:319]: How about `Buffer truncation value "${n}" is out of range: value must be between 0 and ${this.length}`

File: streams/byte_slice_stream.ts (hunk 1 above)
Comment (irbull) [reply streams/byte_slice_stream.ts]: Settled on `Cannot construct ByteSliceStream as start must be >= 0: received ${start}`

File: streams/buffer.ts (hunk 4 above)
Comment (irbull) [reply streams/buffer.ts]: I applied this

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**